modules/enhancer.js

```diff
@@ -2,40 +2,37 @@
 
 var resolveStyles = require('./resolve-styles.js');
 var wrapUtils = require('./wrap-utils.js');
+var objectAssign = require('object-assign');
 
 var enhanceWithRadium = function (ComposedComponent) {
-  var RadiumEnhancer = function () {
-    ComposedComponent.prototype.constructor.call(this);
+  const displayName =
+    ComposedComponent.displayName ||
+    ComposedComponent.name ||
+    'Component';
 
-    if (!this.state) {
-      this.state = {};
-    }
-
-    var radiumInitialState = wrapUtils.getInitialState();
-    Object.keys(radiumInitialState).forEach(function (key) {
-      this.state[key] = radiumInitialState[key];
-    }, this);
-  };
+  class RadiumEnhancer extends ComposedComponent {
+    static displayName = `Radium(${displayName})`;
 
-  RadiumEnhancer.prototype = new ComposedComponent();
-  RadiumEnhancer.prototype.constructor = RadiumEnhancer;
+    constructor (props) {
```

Comment:
also pass context, or just change super to `super(...arguments)` (assuming rest params are supported by babel)